### PR TITLE
fix: scrollTo navigation to right code block

### LINKED ISSUE #986: Information bar isn't navigating to the right code block
#### Describe the bug

There are sections in information bar in studio which are made to navigate to particular part of the code block and view more information about that, but currently on clicking many of them, it only navigates to the end of the code block.
#### How to Reproduce
Steps to reproduce the issue. Attach all resources that can help us understand the issue:

https://github.com/asyncapi/studio/assets/115284013/53b0df95-522e-4855-8788-5229b7ea2cc3

#### Expected behavior

Expected behaviour will be the navigation on the right part of the code block. For instance on clicking schema, the following part should appear 
image

If it is ok, than I will like to work on this bug to solve it.
Thanks!!

### PR DESCRIPTION
**Pull request by Saumya40-codes**
**Description**
scrollTo to a editor line was only working (correctly) for JSON files only.
The fix was to handle the case of YAML file separately. 

https://github.com/asyncapi/studio/assets/115284013/5741f387-cd9c-4aeb-816c-d3cfb7627789

**Related issue(s)** Resolves #986 

Thanks to @satishkumarsajjan and @Vishal2002 for all the feedbacks!!!

### CODE REVIEW

**Review by satishkumarsajjan — APPROVED** [review]

File: apps/studio/src/services/navigation.service.ts [hunk 1]
```diff
@@ -28,9 +28,18 @@ export class NavigationService extends AbstractService {
...
+      const doc = this.svcs.editorSvc;
+      const docType = doc.value.startsWith('asyncapi') ? 'yaml' : 'json';
+
+      let range;
+      if (docType === 'yaml') {
+        range = this.svcs.parserSvc.getRangeForYamlPath('asyncapi', jsonPointer);
+      } else {
+        range = this.svcs.parserSvc.getRangeForJsonPath('asyncapi', jsonPointer);
+      }
+
```
Comment (princerajpoot20) [inline apps/studio/src/services/navigation.service.ts:40]:
```suggestion
      const methodName = doc.value.startsWith('asyncapi') ? 'getRangeForYamlPath' : 'getRangeForJsonPath';
      const range = this.svcs.parserSvc[methodName]('asyncapi', jsonPointer);
```

We can use chaining here for a concise code.

File: apps/studio/src/services/parser.service.ts [hunk 2]
```diff
@@ -13,6 +13,8 @@ import type { Diagnostic, ParseOptions } from '@asyncapi/parser/cjs';
 import type { DocumentDiagnostics } from '../state/documents.state';
 import { SchemaParser } from '@asyncapi/parser';
 
+import { getLocationForJsonPath, parseWithPointers } from '@stoplight/yaml';
```
Comment (princerajpoot20) [inline apps/studio/src/services/parser.service.ts:16]:
```suggestion
import { getLocationForJsonPath, parseWithPointers } from '@stoplight/yaml';
```

**Review by princerajpoot20 — CHANGES_REQUESTED** [review]
Great work, @Saumya40-codes ! Just a small suggestion. 
Thanks @satishkumarsajjan and @Vishal2002 for all your inputs.

File: apps/studio/src/services/navigation.service.ts (hunk 1 above)
Comment (Saumya40-codes) [reply apps/studio/src/services/navigation.service.ts]: Thanks @princerajpoot20 !
I've pushed the required changes regarding this.

**Comment by Saumya40-codes** [comment]
Hey @Amzani I've fixed the issue related to failed test (lint script) 

image

**Review by KhudaDad414 — APPROVED** [review]
LGTM

**Comment by KhudaDad414** [comment]
@princerajpoot20 anything from your side?

**Comment by princerajpoot20** [comment]
@KhudaDad414 All good.

**Review by princerajpoot20 — APPROVED** [review]
@Saumya40-codes Great work! 🚀

**Pull request merged**